lib/init-7.scm

```diff
@@ -143,13 +143,21 @@
     (lambda (expr use-env mac-env)
       (f expr mac-env))))
 
-(define er-macro-transformer
+(define er-macro-transformer*
```

Comment:
I decided that only `er-macro-transformer` should keep the old behaviour of complaining when a syntax keyword is used outside of operator position. There are hardly any non-`er` macros in the Chibi standard library anyway.

Reply:
Is that really the Right Thing?  Allowing er-macro names to appear in operand position allows procedure/datum (e.g. keyword) macros on top of ER as well as syntax-case.  The e argument would be an identifier, which can't otherwise happen.

Reply:
You seem to have understood this the wrong way around:  `er-macro-transformer` *will* complain if the `expr` argument is not a pair. This means all but one or two of the macros in the Chibi standard library will continue to give a helpful error message if their keywords appear outside operator position, also including ones written with `syntax-rules` (since `syntax-rules` expands into an `er-macro-transformer`). There’s a new `er-macro-transformer*` which doesn’t do this check explicitly and so can be used to write identifier macros. In addition, `sc-macro-transformer` and `rsc-macro-transformer` also don’t explicitly check their input types. In practice, this means the error message you’ll get for using those macros, if they weren’t intended to be used as identifier macros, will just be a somewhat confusing ‘can’t take car/cdr of a non-pair’ — this affects one or two macros in the Chibi `lib` directory but none in `(chibi)` or the R7RS libraries.

Reply:
I think Daphne made this decision because of the legacy code using `er-macro-transformer` in Chibi, which assumes a list argument for the form to be transformed. While it wouldn't be incorrect to extend the domain of `er-macro-transformer` silently, it would reduce the quality of error messages in Chibi.

Should `er-macro-transformer` be standardized one day, it would make sense to standardize the version that also handles identifier macros, of course.

An alternative to Daphne's approach would be to go through all ER macros in Chibi. That won't take as long as it sounds, I think.

Reply:
Not only the code in Chibi, but the fact that `er-macro-transformer` is certainly the most commonly-used of the low-level macro transformation systems in code outside of Chibi itself.